src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -2505,7 +2506,17 @@ class ArnoldObject : public ArnoldObjectBase
 					}
 					else
 					{
-						IECore::msg( IECore::Msg::Warning, "ArnoldObject::link()", "Attempt to link nonexistent light" );
+						if( !arnoldLight )
+						{
+							// Not aware of any way this could happen
+							IECore::msg( IECore::Msg::Warning, "ArnoldObject::link()", "Attempt to link nonexistent light" );
+						}
+						else
+						{
+							// We have an ArnoldLight, but with an invalid lightShader.
```

Comment:
Looks like we need a warning at https://github.com/GafferHQ/gaffer/pull/4386/files#diff-97e5d5b15fbd1497b1c3c63069de557c831cbe96bdb5fc7b8278d38290eaa31dL2328. We're currently warning for the "have shader but no mesh" case, but not for the "don't have shader" case.

Reply:
Fixed in place.